File: app/src/main/java/com/botijasoftware/utils/ui/FadeMessage.java
package com.botijasoftware.utils.ui;

import com.botijasoftware.utils.Rectanglef;
import com.botijasoftware.utils.Texture;


public class FadeMessage extends Message {
	public FadeMessage(Texture texture, Rectanglef rect, float time) {
		super(texture, rect, time);
	}

	public void Update(float time) {	
		super.Update(time);
		
		if (state == STATE_IN) {
			alpha = elapsedTime / inTime;
		}
		else if (state == STATE_SHOW) {
			alpha = 1.0f;
		}
		else if (state == STATE_OUT) {
			alpha = 1.0f - (elapsedTime / outTime);
		}
	}
	
}

File: app/src/main/java/com/botijasoftware/utils/ui/FadeMessage.kt
package com.botijasoftware.utils.ui

import com.botijasoftware.utils.Rectanglef
import com.botijasoftware.utils.Texture


class FadeMessage(texture: Texture, rect: Rectanglef, time: Float) : Message(texture, rect, time) {

    override fun Update(time: Float) {
        super.Update(time)

        when (state) {
            Message.STATE_IN -> alpha = elapsedTime / inTime
            Message.STATE_SHOW -> alpha = 1.0f
            Message.STATE_OUT -> alpha = 1.0f - elapsedTime / outTime
        }
    }

}
